modules/account-lib/src/coin/hbar/transaction.ts

```diff
@@ -26,189 +25,117 @@ export class Transaction extends BaseTransaction {
-    const sigPair = new proto.SignaturePair();
-    sigPair.pubKeyPrefix = toUint8Array(key.getKeys(true).pub);
-    sigPair.ed25519 = toUint8Array(signature);
-
-    const sigMap = this._hederaTx.sigMap || new proto.SignatureMap();
-    sigMap.sigPair!.push(sigPair);
-    this._hederaTx.sigMap = sigMap;
+    const sigPair = new SignaturePair();
+    sigPair.setPubkeyprefix(toUint8Array(key.getKeys(true).pub));
+    sigPair.setEd25519(toUint8Array(signature));
+
+    const sigMap = this.hederaTx._toProto().getSigmap() || new SignatureMap();
+
+    const innerTx = this.hederaTx._toProto();
+    sigMap.getSigpairList().push(sigPair);
+    innerTx.setSigmap(sigMap);
+    // The inner transaction must be replaced with the new signed transaction
+    this.fromBytes(innerTx.serializeBinary());
+    // Previous signatures are kept so we just add the new signature to the list
     this._signatures.push(signature);
   }
 
+  /**
+   * Replace transaction signatures with the ones from
+   * the inner SDK transaction
+   */
+  private replaceSignatures(): void {
+    this._signatures = this.hederaTx
+      ._toProto()
+      .getSigmap()!
+      .getSigpairList()
+      .map(sigPair => {
+        return toHex(sigPair.getEd25519_asU8());
+      });
+  }
+
+  /**
+   * Get the inner transaction body deserialized in the SDK
+   * proto default format
+   *
+   * @returns {TransactionBody} a transaction body object
+   */
+  txBody(): TransactionBody {
+    return TransactionBody.deserializeBinary(this._hederaTx._toProto().getBodybytes_asU8());
+  }
 
   /** @inheritdoc */
   toBroadcastFormat(): string {
-    return toHex(this.encode(this._hederaTx));
+    return toHex(this.hederaTx.toBytes());
   }
 
   /** @inheritdoc */
   toJson(): TxData {
-    const [acc, time] = this.getTxIdParts();
+    const txData = this.hederaTx._toProto().toObject();
+    const txBody = this.txBody().toObject();
+
+    const acc = stringifyAccountId(txBody.transactionid!.accountid!);
+    const time = stringifyTxTime(txBody.transactionid!.transactionvalidstart!);
     const result: TxData = {
       id: acc + '@' + time,
-      hash: this.getTxHash(), // TODO: Update once hedera-sdk release this functionality BGA-284
-      data: toHex(this._hederaTx.bodyBytes),
-      fee: new BigNumber(this._txBody.transactionFee!.toString()).toNumber(),
+      hash: txData.sigmap ? toHex(this.hederaTx.hash()) : '', // Hash is returned if the transaction was signed
+      data: toHex(this.hederaTx._toProto().getBodybytes_asU8()),
+      fee: new BigNumber(txBody.transactionfee).toNumber(),
       from: acc,
       startTime: time,
-      validDuration: this._txBody.transactionValidDuration!.seconds!.toString(),
-      node: stringifyAccountId(this._txBody.nodeAccountID!),
-      memo: this._txBody.memo,
+      validDuration: txBody.transactionvalidduration!.seconds.toString(),
+      node: stringifyAccountId(txBody.nodeaccountid!),
+      memo: txBody.memo,
     };
```

Comment:
Is this a custom format? Don't they have a type or a way to export the tx details in a JSON format?

Reply:
Yes this is a custom format including the encoded body that isn't included on the SDK JSON.
The toString() method from the SDK returns a JSON with the other fields but includes the signature map and we thought that it could be an issue.